### DISCUSSION TITLE
Extract text from "flattened" PDF?

### DISCUSSION THREAD

**Discussion by jasonharitou** [pypdfium2-team/pypdfium2#310] [Q&A] [answered]
Hi,

I'm attempting to extract text from a PDF, but it's not being extracted. I've tried with multiple other PDFs and the text extracts fine, so there's something particular about this PDF in question.

Could anyone suggest some ways I can debug why the text is not being extracted? For reference the PDF is a single page, version 1.4, Author is Canva, and Security is None.

File itself is 1.8MB.

Any help would be appreciated. Thanks

**Comment by jasonharitou** [comment]
Did a bit more testing with Canva - when a document is "flattened" during the export process, it becomes unreadable by pypdfium2. Please see the two attached files.
[canva-flattened.pdf](https://github.com/pypdfium2-team/pypdfium2/files/15027535/canva-flattened.pdf)
[canva-full.pdf](https://github.com/pypdfium2-team/pypdfium2/files/15027536/canva-full.pdf)

**Comment by mara004** [comment, marked as answer]
AFAICS, the `canva_flattened` PDF just doesn't have a text layer, so there is nothing for a pdf engine to extract. You could use an OCR tool like ocrmypdf to change that, though.
Also note that `canva_full` is a vector PDF, whereas `canva_flattened` merely contains a rasterized image of the former.
For this reason, I recommend just not using this "flattening", as it evidently rasterizes, which is inelegant (maybe you've noticed the file size of `canva_flattened` is way higher than of `canva_full`).

**Reply by jasonharitou** [reply]
Yeah I noticed the full version was massive! Thanks for your answer, OCR tools here I come…

**Reply by mara004** [reply]
(_note for readers:_ canva's flattening should not be confused with PDF form field flattening as provided by pdfium)
